feat: add C ndarray interface and refactor implementation for `stats/base/svariancewd`

Vinit-Pandit opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   {{TODO: add description describing what this pull request does}}

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add `eslint/rules/eol-open-bracket-spacing`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4061
Private-ref: https://github.com/stdlib-js/todo/issues/2324
Reviewed-by: Athan Reines <kgryte@gmail.com>

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: failed
---

Vinit-Pandit committed:
Merge remote-tracking branch 'upstream/develop' into develop

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: failed
---

Vinit-Pandit committed:
Merge remote-tracking branch 'upstream/develop' into develop

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Vinit-Pandit committed:
Merge remote-tracking branch 'upstream/develop' into develop

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Vinit-Pandit committed:
Merge remote-tracking branch 'upstream/develop' into develop

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: failed
---

Vinit-Pandit committed:
Merge remote-tracking branch 'upstream/develop' into develop

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: failed
---

Vinit-Pandit committed:
feat: add C ndarray interface and refactor implementation for stats/base/svariancewd

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/examples/c/example.c:
```diff
@@ -17,21 +17,20 @@
 int main( void ) {
 	// Create a strided array:
-	float x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 };
+	float x[] = { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f };
```
```suggestion
	const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f };
```

Vinit-Pandit committed:
clean up

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -117,18 +116,16 @@ The function has the following parameters:
+-   **strideX**: stride length for `x`.
 
-The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to compute the [variance][variance] of every other element in `x`,
+The `N` and stride parameters determine which elements in `x` are accessed at runtime. For example, to compute the [variance][variance] of every other element in `x`,
```
```suggestion
The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to compute the [variance][variance] of every other element in `x`,
```

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -138,45 +135,40 @@ Note that indexing is relative to the first index. To introduce an offset, use [
+-   **offsetX**: starting index for `x`.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to calculate the [variance][variance] for every other value in `x` starting from the second value
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the offset parameter supports indexing semantics based on a starting index. For example, to calculate the [variance][variance] for every other element in `x` starting from the second element
```
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to calculate the [variance][variance] for every other element in `x` starting from the second element
```

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -224,10 +210,131 @@ console.log( v );
+
+The function accepts the following arguments:
+
+-   **N**: `[In] CBLAS_INT` number of indexed elements.
+-   **correction**: `[In] FLOAT` degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [variance][variance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the [variance][variance] of a population, setting this parameter to `0` is the standard choi ...
+-   **X**: `[in] float*` input array.
+-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
```
```suggestion
-   **N**: `[in] CBLAS_INT` number of indexed elements.
-   **correction**: `[in] FLOAT` degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [variance][variance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the [variance][variance] of a population, setting this parameter to `0` is the standard choice (i.e., the provided array contains data constituting an entire population). When computing the unbiased sample [variance][variance], setting this parameter to `1` is the standard choice (i.e., the provided array contains data sampled from a larger population; this is commonly referred to as Bessel's correction).
-   **X**: `[in] float*` input array.
-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
```

Vinit-Pandit committed:
fixes in readme

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit committed:
fix: fixing cpp error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit commented:
/stdlib update-copyright-years

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -117,18 +116,16 @@ The function has the following parameters:
--   **stride**: index increment for `x`.
+-   **strideX**: stride length for `x`.
 
 The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to compute the [variance][variance] of every other element in `x`,
```
```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to compute the [variance][variance] of every other element in `x`,
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
````diff
@@ -138,45 +135,40 @@ Note that indexing is relative to the first index. To introduce an offset, use [
 
 ```javascript
 var Float32Array = require( '@stdlib/array/float32' );
 var floor = require( '@stdlib/math/base/special/floor' );
````
```suggestion

```

aayush0325 replied:
we don't need this import anymore

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -224,10 +210,131 @@ console.log( v );
+-   **X**: `[in] float*` input array.
+-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
+
+The `N` and stride parameters determine which elements in `x` are accessed at runtime. For example, to compute the [variance][variance] of every other element in `x`,
```
```suggestion
```

aayush0325 replied:
we don't need this in the C interface documentation

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/benchmark/benchmark.ndarray.js:
```diff
@@ -21,14 +21,20 @@
+
+var option = {
+	'dtype': 'float32'
+};
```
```suggestion
var options = {
	'dtype': 'float32'
};
```
just updating this as a matter of convention

aayush0325 replied:
same comment for all other javascript benchmarks

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/examples/index.js:
```diff
@@ -18,18 +18,14 @@
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( (randu()*100.0) - 50.0 );
-}
+x = discreteUniform( 10, -50, 50, {
```
```suggestion
var x = discreteUniform( 10, -50, 50, {
```

aayush0325 replied:
we can declare and assign here

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/include/stdlib/stats/base/svariancewd.h:
```diff
@@ -31,7 +31,12 @@ extern "C" {
+float API_SUFFIX(stdlib_strided_svariancewd)( const CBLAS_INT N, const float correction, const float *X, const CBLAS_INT strideX );
+
+/**
+* Computes the variance of a single-precision floating-point strided array using Welford's algorithm and using alternative indexing semantics..
```
```suggestion
* Computes the variance of a single-precision floating-point strided array using Welford's algorithm and using alternative indexing semantics.
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/src/main.c:
```diff
@@ -64,10 +65,10 @@
 */
-float stdlib_strided_svariancewd( const int64_t N, const float correction, const float *X, const int64_t stride ) {
+float API_SUFFIX(stdlib_strided_svariancewd)( const CBLAS_INT N, const float correction, const float *X, const CBLAS_INT strideX ) {
 	float delta;
```
this implementation is not what we are looking for, see #4240, notice how in main.c we're getting the offset according to `N` and `strideX` using `stdlib/strided/base/stride2offset` and then making a call to the ndarray interface, you'll need to include `stdlib/strided/base/stride2offset.h` and hence add it in manifest.json too

aayush0325 replied:
the function should look like 

```c
const CBLAS_INT ox = stdlib_base_stride2offset( N, strideX );
return API_SUFFIX(stdlib_strided_svariancewd_ndarray)( N, correction, X, strideX, ox ); 
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/src/addon.c:
```diff
@@ -32,14 +32,33 @@
 * @param info   callback data
 * @return       Node-API value
 */
+float API_SUFFIX(stdlib_strided_svariancewd)( const CBLAS_INT N, const float correction, const float *X, const CBLAS_INT strideX );
```
```suggestion
```
we don't need this here

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/src/main.c:
```diff
@@ -17,6 +17,7 @@
 
 #include "stdlib/stats/base/svariancewd.h"
+#include "stdlib/blas/base/shared.h"
 #include <stdint.h>
```
```suggestion
```

we dont need to include this header anymore as `int64_t` is not being used

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/manifest.json:
```diff
@@ -72,7 +79,26 @@
+      ],
+      "libraries": [
+        "-lm"
+      ],
```
```suggestion
      "libraries": [],
```
we don't need this anymore, same for all other instances in this file

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/lib/ndarray.native.js:
```diff
@@ -32,27 +31,21 @@ var addon = require( './svariancewd.native.js' );
 *
 * @example
 * var Float32Array = require( '@stdlib/array/float32' );
 * var floor = require( '@stdlib/math/base/special/floor' );
```
```suggestion
```
we don't need to import this module anymore

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/lib/ndarray.js:
```diff
@@ -70,21 +70,20 @@ var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
 *
 * @example
 * var Float32Array = require( '@stdlib/array/float32' );
 * var floor = require( '@stdlib/math/base/special/floor' );
```
```suggestion
```
we don't need to import this module anymore

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/lib/index.js:
```diff
@@ -28,9 +28,8 @@
-* var v = svariancewd( N, 1, x, 1 );
+* var v = svariancewd( x.length, 1, x, 1 );
 * // returns ~4.3333
 *
```
we don't need to import `floor` here

aayush0325 reviewed (changes requested):
left an initial round of comments

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/lib/svariancewd.js:
please refer to an existing implementation to get an idea of what we're looking for here! similar to C implementation we need to use `stride2offset` here

aayush0325 replied:
your function should look like 
```javascript
return ndarray( N, x, strideX, stride2offset( N, strideX ) );
```
and you need to import ndarray and stride2offset by
```javascript
var stride2offset = require( '@stdlib/strided/base/stride2offset' );
var ndarray = require( './ndarray.js' );
```

aayush0325 replied:
you can refer to [this](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/base/dmin/lib/dmin.js)

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/src/main.c:
```diff
@@ -95,7 +96,47 @@ float stdlib_strided_svariancewd( const int64_t N, const float correction, const
+
+
+/**
+* Computes the variance of a single-precision floating-point strided array using Welford's algorithm.
```
```suggestion
* Computes the variance of a single-precision floating-point strided array using Welford's algorithm using alternative indexing semantics.
```

Vinit-Pandit commented:
Thank You @aayush0325 for your review ,  one more question is where i can find all the latest convention of project , means is it is written somewhere ?  or just have to look other updated pacakges to get idea of all this updates we are making ?

aayush0325 commented:
@MeastroZI check stats/base/dmax!

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/docs/repl.txt:
```diff
@@ -31,7 +31,7 @@
 
-    stride: integer
+    strideX: integer
         Index increment.
```
```suggestion
        Stride Length.
```
same for all other instances in this file

Vinit-Pandit committed:
Apply suggestions from code review

Co-authored-by: Aayush Khanna <96649223+aayush0325@users.noreply.github.com>
Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit committed:
fix: fixing mistakes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/examples/index.js:
```diff
@@ -18,18 +18,14 @@
+var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
 var svariancewd = require( './../lib' );
 
 var x;
```
```suggestion
```

Vinit-Pandit committed:
fix : removing multiple variable declaration

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/examples/index.js:
```diff
@@ -18,18 +18,13 @@
-var Float32Array = require( '@stdlib/array/float32' );
+var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
 var svariancewd = require( './../lib' );
 
```
```suggestion
```

Vinit-Pandit committed:
style: removing extra line

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -202,18 +193,12 @@ var v = svariancewd.ndarray( N, 1, x, 2, 1 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( (randu()*100.0) - 50.0 );
-}
+x = discreteUniform( 10, -50, 50 );
```
please set the datatype to be float32 explicitly

aayush0325 replied:
this example should match `examples/index.js`

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svariancewd/benchmark/benchmark.native.js:
```diff
@@ -49,12 +51,8 @@ var opts = {
-	for ( i = 0; i < x.length; i++ ) {
-		x[ i ] = ( randu()*20.0 ) - 10.0;
-	}
+	x = uniform( len, -10, 10, options);
```
```suggestion
	var x = uniform( len, -10, 10, options );
```
we can simply declare and assign here, same comment for all other javascript benchmark files

aayush0325 replied:
```suggestion
	var x = uniform( len, -10, 10, options );
```
spacing is also off here, please check for this in other JavaScript benchmarks too!

aayush0325 reviewed (changes requested):
minor changes needed, this PR will be in good shape once these are implemented!

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -202,18 +193,12 @@ var v = svariancewd.ndarray( N, 1, x, 2, 1 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( (randu()*100.0) - 50.0 );
-}
+x = discreteUniform( 10, -50, 50 );
```
```suggestion
x = discreteUniform( 10, -50, 50, {
	'dtype': 'float32'
});
```

aayush0325 replied:
can you please do something like 
```javascript
var x = discreteUniform( 10, -50, 50, {
	'dtype': 'float32'
});
```
i.e. declare and assign at the same time

aayush0325 replied:
in cases where the calculation is **not complex** we can declare and assign
```javascript
var x = someFunction();
```

is better than

```javascript
var x;

x = someFunction();
```

Vinit-Pandit committed:
fix: set the datatype to be float32 explicitly

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit committed:
fix: declare and assign variables

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 reviewed (approved):
lgtm

kgryte commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -224,10 +209,130 @@ console.log( v );
+
+#### stdlib_strided_svariancewd( N, correction, \*X, strideX )
+
+Computes the [variance][variance] of a single-precision floating-point strided array `x` using Welford's algorithm.
```
```suggestion
Computes the [variance][variance] of a single-precision floating-point strided array using Welford's algorithm.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/svariancewd/README.md:
```diff
@@ -224,10 +209,130 @@ console.log( v );
+The function accepts the following arguments:
+
+-   **N**: `[in] CBLAS_INT` number of indexed elements.
+-   **correction**: `[in] FLOAT` degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [variance][variance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the [variance][variance] of a population, setting this parameter to `0` is the standard choi ...
```
```suggestion
-   **correction**: `[in] float` degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [variance][variance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the [variance][variance] of a population, setting this parameter to `0` is the standard choice (i.e., the provided array contains data constituting an entire population). When computing the unbiased sample [variance][variance], setting this parameter to `1` is the standard choice (i.e., the provided array contains data sampled from a larger population; this is commonly referred to as Bessel's correction).
```

kgryte commented on lib/node_modules/@stdlib/stats/base/svariancewd/docs/repl.txt:
```diff
@@ -1,9 +1,9 @@
     using Welford's algorithm.
 
-    The `N` and `stride` parameters determine which elements in `x` are accessed
+    The `N` and stride parameters determine which elements in x are accessed
```
```suggestion
    The `N` and stride parameters determine which elements in the strided array are accessed
```

You will likely need to rewrap the lines. The point of removing references to `x` is to avoid copy-paste errors in future packages.

kgryte commented on lib/node_modules/@stdlib/stats/base/svariancewd/include/stdlib/stats/base/svariancewd.h:
```diff
@@ -31,7 +31,12 @@ extern "C" {
+float API_SUFFIX(stdlib_strided_svariancewd)( const CBLAS_INT N, const float correction, const float *X, const CBLAS_INT strideX );
+
+/**
+* Computes the variance of a single-precision floating-point strided array using Welford's algorithm and using alternative indexing semantics.
```
```suggestion
* Computes the variance of a single-precision floating-point strided array using Welford's algorithm and alternative indexing semantics.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/svariancewd/src/main.c:
```diff
@@ -64,10 +65,26 @@
+	return API_SUFFIX(stdlib_strided_svariancewd_ndarray)( N, correction, X, strideX, ox );
+}
+
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/stats/base/svariancewd/src/main.c:
```diff
@@ -64,10 +65,26 @@
+
+
+/**
+* Computes the variance of a single-precision floating-point strided array using Welford's algorithm using alternative indexing semantics.
```
```suggestion
* Computes the variance of a single-precision floating-point strided array using Welford's algorithm and alternative indexing semantics.
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (changes requested):
Apart from reflowing the text in the REPL text file, this is looking good to me.

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into nd_array_svariancewd

Vinit-Pandit committed:
fix: fixing repl txt

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit committed:
style: fixing space in manifest json

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit committed:
fix: add trailing space in repl txt

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 reviewed (approved):
LGTM
